iterator-test-harness/src/test/java/org/apache/accumulo/iteratortest/WholeRowIteratorTest.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.accumulo.iteratortest.junit4.BaseJUnit4IteratorTest;
+import org.apache.accumulo.iteratortest.testcases.IteratorTestCase;
+import org.apache.hadoop.io.Text;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Framework tests for {@link WholeRowIterator}.
+ */
+public class WholeRowIteratorTest extends BaseJUnit4IteratorTest {
+
+  @Parameters
+  public static Object[][] parameters() {
+    IteratorTestInput input = getIteratorInput();
+    IteratorTestOutput output = getIteratorOutput();
+    List<IteratorTestCase> tests = IteratorTestCaseFinder.findAllTestCases();
+    return BaseJUnit4IteratorTest.createParameters(input, output, tests);
+  }
+
+  private static final TreeMap<Key,Value> INPUT_DATA = createInputData();
+  private static final TreeMap<Key,Value> OUTPUT_DATA = createOutputData();
+
+  private static TreeMap<Key,Value> createInputData() {
+    TreeMap<Key,Value> data = new TreeMap<>();
+
+    data.put(new Key("1", "", "a"), new Value("1a".getBytes()));
+    data.put(new Key("1", "", "b"), new Value("1b".getBytes()));
+    data.put(new Key("1", "a", "a"), new Value("1aa".getBytes()));
+    data.put(new Key("1", "a", "b"), new Value("1ab".getBytes()));
+    data.put(new Key("1", "b", "a"), new Value("1ba".getBytes()));
+
+    data.put(new Key("2", "a", "a"), new Value("2aa".getBytes()));
+    data.put(new Key("2", "a", "b"), new Value("2ab".getBytes()));
+    data.put(new Key("2", "a", "c"), new Value("2ac".getBytes()));
+    data.put(new Key("2", "c", "c"), new Value("2cc".getBytes()));
+
+    data.put(new Key("3", "a", ""), new Value("3a".getBytes()));
+
+    data.put(new Key("4", "a", "b"), new Value("4ab".getBytes()));
+
+    data.put(new Key("5", "a", "a"), new Value("5aa".getBytes()));
+    data.put(new Key("5", "a", "b"), new Value("5ab".getBytes()));
+    data.put(new Key("5", "a", "c"), new Value("5ac".getBytes()));
+    data.put(new Key("5", "a", "d"), new Value("5ad".getBytes()));
+
+    data.put(new Key("6", "", "a"), new Value("6a".getBytes()));
+    data.put(new Key("6", "", "b"), new Value("6b".getBytes()));
+    data.put(new Key("6", "", "c"), new Value("6c".getBytes()));
+    data.put(new Key("6", "", "d"), new Value("6d".getBytes()));
+    data.put(new Key("6", "", "e"), new Value("6e".getBytes()));
+    data.put(new Key("6", "1", "a"), new Value("61a".getBytes()));
+    data.put(new Key("6", "1", "b"), new Value("61b".getBytes()));
+    data.put(new Key("6", "1", "c"), new Value("61c".getBytes()));
+    data.put(new Key("6", "1", "d"), new Value("61d".getBytes()));
+    data.put(new Key("6", "1", "e"), new Value("61e".getBytes()));
+
+    return data;
+  }
+
+  private static TreeMap<Key,Value> createOutputData() {
+    TreeMap<Key,Value> data = new TreeMap<>();
+
+    Text row = null;
+    List<Key> keys = new ArrayList<>();
+    List<Value> values = new ArrayList<>();
+
+    // Generate the output data from the input data
+    for (Entry<Key,Value> entry : INPUT_DATA.entrySet()) {
```

Comment:
Using [RowIterator](https://accumulo.apache.org/1.6/apidocs/org/apache/accumulo/core/client/RowIterator.html) may simplify this code a bit.